File: Source/Game/Podium/PodiumManager.cs
using WCSharp.Shared.Data;

public static class PodiumManager
{
    public static Point[] PodiumSpots;

    public static void Initialize()
    {
        SetPodiumPositions();
    }

    public static void BeginPodiumEvents()
    {
        switch (Gamemode.CurrentGameMode)
        {
            case GameMode.Standard:
                StandardPodium.BeginPodiumActions();
                break;

            case GameMode.SoloTournament:
                SoloPodium.BeginPodiumActions();
                break;

            case GameMode.TeamTournament:
                TeamPodium.BeginPodiumActions();
                break;
        }
    }

    private static void SetPodiumPositions()
    {
        PodiumSpots = new[]
{
            Regions.Podium_1.Center,
            Regions.Podium_2.Center,
            Regions.Podium_3.Center
        };
    }
}

File: src/Game/Podium/PodiumManager.ts
import { CurrentGameMode } from 'src/Gamemodes/CurrentGameMode'
import { GameMode } from 'src/Gamemodes/GameModeEnum'
import { SoloPodium } from './SoloPodium'
import { StandardPodium } from './StandardPodium'
import { TeamPodium } from './TeamPodium'

export class PodiumManager {
    public static BeginPodiumEvents = () => {
        switch (CurrentGameMode.active) {
            case GameMode.Standard:
                StandardPodium.BeginPodiumActions()
                break

            case GameMode.SoloTournament:
                SoloPodium.BeginPodiumActions()
                break

            case GameMode.TeamTournament:
                TeamPodium.BeginPodiumActions()
                break
        }
    }
}
